packages/apputils/src/sessioncontext.tsx

```diff
@@ -1265,6 +1277,13 @@ export namespace SessionContext {
    * Get the default kernel name given select options.
    */
   export function getDefaultKernel(options: IKernelSearch): string | null {
+    const { preference } = options;
+    const { shouldStart } = preference;
+
+    if (shouldStart === false) {
+      return null;
+    }
+
```

Comment:
I added this to make public `getDefaultKernel` be backward compatible.
Honestly, I don't like this change... It is very hard to understand this code without knowing history.